relayer/relays/execution/main.go

```diff
@@ -145,6 +146,22 @@ func (r *Relay) Start(ctx context.Context, eg *errgroup.Group) error {
 					logger.Warn("inbound message outdated, just skipped")
 					continue
 				}
+				nextBlockNumber := new(big.Int).SetUint64(ev.Raw.BlockNumber + 1)
+
+				blockHeader, err := ethconn.Client().HeaderByNumber(ctx, nextBlockNumber)
+				if err != nil {
+					return fmt.Errorf("get block header: %w", err)
+				}
+
+				// ParentBeaconRoot in https://eips.ethereum.org/EIPS/eip-4788 from Deneb onward
+				err = beaconHeader.SyncExecutionHeader(ctx, *blockHeader.ParentBeaconRoot)
+				if err == header.ErrBeaconHeaderNotFinalized {
+					logger.Warn("beacon header not finalized, just skipped")
+					continue
+				}
```

Comment:
Just continue and wait here until that execution header of the event log is finalized.